content/blog/kubernetes-docker-support.fr.md

```diff
@@ -0,0 +1,140 @@
+---
+Title: Kubernetes ne supporte plus Docker. Voici pourquoi ce n'est pas grave.
+Date: 2020-12-03
+Category: Kubernetes
+Summary: Kubernetes stop le support de Docker à partir de la version 1.20, voici pourquoi ce n'est absolument pas grave et même une bonne chose
+Author: Romain Guichard
+image: images/thumbnails/kubernetes.png
+imgSocialNetwork: images/og/kubernetes-docker-support.png
+lang: fr
+---
+
+
+[C'est l'info du jour qui fait trembler tout le monde : Kubernetes va déprécier
+le support de Docker](https://github.com/kubernetes/kubernetes/pull/94624)
+à partir de la version `1.20` dont la sortie est prévue
+dans quelques jours/semaines.
+
+Catastrophe.
+
+Ou pas.
+
+Revenons tout d'abord sur ce qu'est Docker et sur sa relation avec Kubernetes.
+
+
+# Docker, le mauvais élève des container runtimes
+
+Docker est une stack technologique fournissant plusieurs mécanismes pour faire
+fonctionner vos conteneurs. Il permet de construire des images, de créer des
+volumes, de créer des réseaux, de démarrer des conteneurs et grâce à son
+intégration avec Swarm, permet de se transformer en orchestrateur et de gérer
+des clusters d'hôtes Docker.
+
+Kubernetes en revanche est simplement un orchestrateur de conteneurs. Il ne
+build pas d'image, ne crée pas de réseau, ne crée pas de volume et ne démarre
+même pas de conteneurs. Pour toutes ces actions, Kubernetes a défini des
+interfaces permettant à n'importe quel logiciel tiers d'effectuer ces actions à
+sa place. Ces interfaces sont actuellement au nombre de trois :
+
+- [Container Runtime Interface (CRI)](https://github.com/kubernetes/cri-api)
+- [Container Network Interface
+  (CNI)](https://github.com/containernetworking/cni)
+- [Container Storage Interface
+  (CSI)](https://github.com/container-storage-interface/)
+
+Ces interfaces vous permettent d'utiliser Weave, Calico ou Cilium comme plugin
+réseau, car ces solutions sont compatibles avec CNI. Elles vous permettent
+d'utiliser Ceph, AWS EBS ou OpenStack Manila comme backend pour vos Persistent
+Volumes car ces solutions sont compatibles avec CSI. Kubernetes n'a donc pas
+besoin que ces fonctions soient supportées par Docker, Kubernetes n'a besoin
+que d'une seule chose de la part de Docker : être une **container runtime**. Et
+malheureusement Docker n'est pas compatible avec CRI, on doit utiliser un
+composant appelé **dockershim** pour faire travailler Docker et Kubernetes
+ensemble. Ce n'est pas grave en soit, mais c'est un développement spécifique à
+prévoir uniquement pour faire fonctionner Docker, alors que des implémentations
+compatibles existent. C'est le support de ce composant qui est en réalité
+arrêté.
+
+Revenons sur ce qu'est une container runtime. [Vous pouvez aussi retrouver notre
+article](https://particule.io/blog/container-runtime/).
+
+# Les container runtimes
+
```

Comment:
add a link somewhere to the container runtime article

Reply:
It's just there, line 59